content/4_Gold/TopoSort.mdx

```diff
@@ -61,121 +61,145 @@ involving BFS.
 </Resources>
 
 <LanguageSection>
-
 <CPPSection>
 
 ```cpp
-#include <bits/stdc++.h>
-using namespace std;
+#include <algorithm>
+#include <iostream>
+#include <vector>
 
-#define pb push_back
+using std::cout;
+using std::endl;
+using std::vector;
 
-int N;                                // Number of nodes
-vector<int> graph[100000], top_sort;  // Assume that this graph is a DAG
-bool visited[100000];
+vector<int> top_sort;
+vector<vector<int>> graph;
+vector<bool> visited;
 
 void dfs(int node) {
-	for (int i : graph[node]) {
-		if (!visited[i]) {
-			visited[i] = true;
-			dfs(i);
+	for (int n : graph[node]) {
```

Comment:
whats wrong with $i$?

i know literally 0 person who uses $n$ instead and it can be confused with the $n$ in the problem statement (some people like to declare $n$ globally.

Reply:
n is for neighbor, and honestly if you're confusing `n` with the problem one that's on you given that the scope this n is used in is so tiny

Reply:
change it to child, $i$, or next_node please

Reply:
> honestly if you're confusing n with the problem one that's on you given that the scope this n is used in is so tiny

Reply:
well i changed it
honestly it was probably better since some of code already used `next`